modules/bullet/space_bullet.cpp

```diff
@@ -675,30 +668,43 @@ void SpaceBullet::check_ghost_overlaps() {
 		const btAlignedObjectArray<btCollisionObject *> ghostOverlaps = area->get_bt_ghost()->getOverlappingPairs();
 
 		// For each overlapping
-		for (i = ghostOverlaps.size() - 1; 0 <= i; --i) {
+		for (int i = ghostOverlaps.size() - 1; 0 <= i; --i) {
 
 			if (ghostOverlaps[i]->getUserIndex() == CollisionObjectBullet::TYPE_AREA) {
 				if (!static_cast<AreaBullet *>(ghostOverlaps[i]->getUserPointer())->is_monitorable())
 					continue;
 			} else if (ghostOverlaps[i]->getUserIndex() != CollisionObjectBullet::TYPE_RIGID_BODY)
 				continue;
 
-			otherObject = static_cast<RigidCollisionObjectBullet *>(ghostOverlaps[i]->getUserPointer());
+			RigidCollisionObjectBullet *otherObject = static_cast<RigidCollisionObjectBullet *>(ghostOverlaps[i]->getUserPointer());
 
 			bool hasOverlap = false;
 
 			// For each area shape
-			for (y = area->get_shape_count() - 1; 0 <= y; --y) {
-				if (!area->get_bt_shape(y)->isConvex())
+			for (int y = area->get_shape_count() - 1; 0 <= y; --y) {
+				btConvexShape *area_shape = static_cast<btConvexShape *>(area->get_bt_shape(y));
+
+				if (!area_shape) {
```

Comment:
This check is not correct here, even if it fix the problem. The function above should always return a shape and if it doesn't do that, it means there is a bug somewhere else.

I'll check it this weekend, and thanks you to look on it but feel free to check / Fix it if you want :)

Reply:
Thanks you too